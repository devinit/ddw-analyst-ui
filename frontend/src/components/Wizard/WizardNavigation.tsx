import React, { FC } from 'react';
import { Nav } from 'react-bootstrap';
import styled from 'styled-components';

// some strange phenomenon is happening when Navs are wrapped in a Card - a calls is added to them
const StyledNav = styled(Nav)`
  &.nav {
    border-radius: inherit !important;
    margin: inherit !important;
    padding: inherit !important;
  }
`;
type ComponentProps = {
  children?: React.ReactNode;
};

const WizardNavigation: FC<ComponentProps> = ({ children }) => {
  return (
    <div className="wizard-navigation">
      <StyledNav variant="pills" className="">
        {children}
      </StyledNav>
    </div>
  );
};

export { WizardNavigation };
