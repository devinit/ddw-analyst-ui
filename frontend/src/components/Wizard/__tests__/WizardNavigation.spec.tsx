/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardNavigation } from '../WizardNavigation';

describe('WizardNavigation', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardNavigation />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardNavigation>
        <div>My Child</div>
      </WizardNavigation>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
