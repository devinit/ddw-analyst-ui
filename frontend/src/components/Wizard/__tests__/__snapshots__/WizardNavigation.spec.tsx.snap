// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardNavigation renders the default structure correctly 1`] = `
.c0.nav {
  border-radius: inherit!important;
  margin: inherit!important;
  padding: inherit!important;
}

<div
  className="wizard-navigation"
>
  <div
    className="c0 nav nav-pills"
    onKeyDown={[Function]}
  />
</div>
`;

exports[`WizardNavigation renders with children 1`] = `
.c0.nav {
  border-radius: inherit!important;
  margin: inherit!important;
  padding: inherit!important;
}

<div
  className="wizard-navigation"
>
  <div
    className="c0 nav nav-pills"
    onKeyDown={[Function]}
  >
    <div>
      My Child
    </div>
  </div>
</div>
`;
